src/protocol/http/org/apache/jmeter/protocol/http/curl/BasicCurlParser.java

```diff
@@ -252,4 +821,232 @@ public Request parse(String commandLine) {
         }
         return result.toArray(new String[result.size()]);
     }
+    /**
+    *
+    * Set the username , password and baseurl of authorization
+    *
+    * @param authorizationStr the username and password of authorization
+    * @param url              the baseurl of authorization
+    * @param authorization    the object of authorization
+    */
+   public void setAuthUserPasswd(String authorizationStr, String url, Authorization authorization) {
+       String[] authorizationParameters = authorizationStr.split(":");
+       authorization.setUser(authorizationParameters[0].trim());
+       authorization.setPass(authorizationParameters[1].trim());
+       authorization.setURL(url);
+   }
+
+   /**
+    *
+    * Set the mechanism of authorization
+    *
+    * @param mechanism     the mechanism of authorization
+    * @param authorization the object of authorization
+    */
+   private void setAuthMechanism(String mechanism, Authorization authorization) {
+       switch (mechanism) {
+       case "basic":
+           authorization.setMechanism(Mechanism.BASIC);
+           break;
+       case "digest":
+           authorization.setMechanism(Mechanism.DIGEST);
+           break;
+       default:
+           break;
+       }
+   }
+
+   /**
+    *
+    * Set the parameters of proxy server in http request advanced
+    *
+    * @param request         http request
+    * @param proxyServerParameters the parameters of proxy server
+    *
+    */
+   private void setProxyServer(Request request, String proxyServerParameters) {
+       if (!proxyServerParameters.contains("://")) {
+           proxyServerParameters = "http://" + proxyServerParameters;
+       }
+       URI uriProxy = null;
+       try {
+           uriProxy = new URI(proxyServerParameters);
+           request.setProxyServer("scheme", uriProxy.getScheme());
+           Optional<String> userInfoOptional = Optional.ofNullable(uriProxy.getUserInfo());
+           if (userInfoOptional.isPresent()) {
+               String userinfo = userInfoOptional.get();
+               if (userinfo.contains(":")) {
+                   String[] userInfo = userinfo.split(":");
+                   request.setProxyServer("username", userInfo[0]);
+                   request.setProxyServer("password", userInfo[1]);
+               }
+           }
+           Optional<String> hostOptional = Optional.ofNullable(uriProxy.getHost());
+           if (hostOptional.isPresent()) {
+               request.setProxyServer("servername", hostOptional.get());
+           }
+           if (uriProxy.getPort() != -1) {
+               request.setProxyServer("port", String.valueOf(uriProxy.getPort()));
+           } else {
+               request.setProxyServer("port", "1080");
```

Comment:
Why is this port number `1080`? Most of the proxies I have seen are at different ports. Is this curl specific?

Reply:
if the user doesn't specify the port number, it will be 1080(default number).  And user can also uses their own port number like "curl 'http://jmeter.apache.org/' -x 'https://aa:bb@example.com:**8042**'"

Reply:
`-x, --proxy <[protocol://][user@password]proxyhost[:port]>`
 Use the specified HTTP proxy. 
 If the port number is not specified, it is assumed at port 1080.

Reply:
No problem here. I only was curious about the -- to me -- strange proxy port.